nova_vm/src/heap/heap_bits.rs

```diff
@@ -1063,3 +1072,38 @@ pub(crate) fn sweep_heap_elements_vector_descriptors<T>(
         descriptors.insert(new_key, descriptor);
     }
 }
+
+#[cfg(feature = "array-buffer")]
+pub(crate) fn sweep_data_view_side_table_values<V: Copy>(
+    // TODO: Figure out a generic way to handle all types which wrap `BaseIndex`
+    side_table: &mut AHashMap<DataView, V>,
+    compactions: &CompactionList,
+    marks: &[bool],
```

Comment:
**note:** This is absolutely a working choice. I had a vague idea of doing this via the CompactionList by figuring out which shift we get "normally", and if after that and the next highest shift we're overwritten.

eg. Let's say we're index 0 and we've been GC'd. We find no shift index for ourselves in `compactions` and the first index in `compactions` should then be `index: 1, shift: 1` (I think we get this even if we're the only DataView but I may be wrong on this one). Because we found no shift index for ourselves, we get a shift number of 0 from the `unwrap_or` logic. Our post-shift index is then still 0. Now we look at the next shift group's lowest point: `index: 1, shift: 1` means that it shifts down to 0. This is equal or lower to our shifted index, meaning it overwrote us. Hence, we are deleted. We should return `None` to signal that.

For the DataView at index 1 it finds a shift of 1, and the next compaction group is let's say 5 with shift 2. Its post-shift index is 0, and the next compaction group goes down to 3 so it isn't overridden. It can return its shift (or post-shift index?) directly.

For the data view at index 4, it finds a shift of 1, and again the next compaction group is 5 with shift 2. Its post-shift index is 3. Next group shifts down to 3 as well: This one is finished and should return `None`.

Reply:
I'll leave that for you if you want to. Sounds doable but like it would be easier to just let you do it later 😉